### PR TITLE
Read config from different sources

### PR DESCRIPTION
**Pull request by zprogd**
If defined --config=<config_file>, use it.
Else if defined $ENV{PAKKET_CONFIG_FILE}, use it
Else choose first existing of /env/pakket.*,  ~/.pakket.* and use it.

### CODE REVIEW

File: lib/Pakket/Config.pm [hunk 1]
```diff
@@ -46,7 +52,7 @@ sub read_config {
     });
 
     my %cfg;
-    foreach my $config_chunk ( @{$config} ) {
+    if ( my $config_chunk = $config->[0] ) {
```
Comment (xsawyerx) [inline lib/Pakket/Config.pm]: Why is this hard-coded to `0`?

I can setting the files to `map("/etc/pakket.$_", @{...} ), map("$prefix_path.$_", @{...} )`, but wouldn't the first be `0` and the second `1`?

Comment (zprogd) [inline lib/Pakket/Config.pm]: Yes, sorry, wrong arrangement, home directory should go first, then /etc/.
I'll change them.

Comment (zprogd) [inline lib/Pakket/Config.pm]: fixed

Comment (xsawyerx) [inline lib/Pakket/Config.pm]: That's a fine fix, but that's not what I meant. :)

**Pull request closed without merging**